routing_plugin.go

```diff
@@ -104,3 +120,433 @@ func ReleaseRoutingPlugin(ctx context.Context) {
+
+		nodeInfo, err := getNodeInfo(
+			ctx, r.lnd.Client, node, privateEdges,
+		)
+		if err != nil {
+			return err
+		}
+
+		nodes[node] = nodeInfo
+	}
+
+	// Save MC state.
+	entries, err := r.lnd.Router.QueryMissionControl(ctx)
+	if err != nil {
+		return err
+	}
+	r.mcState = entries
+	log.Debugf("Saved MC state: %v", spew.Sdump(r.mcState))
+
+	// maxFeePerNode assigns the maximum fees that would be paid through
+	// selected nodes to the target node for the target amount.
+	maxFeePerNode := make(map[route.Vertex]int64)
+	totalCapacityPerNode := make(map[route.Vertex]btcutil.Amount)
+
+	amtMsat := int64(amt * 1000)
+	for nodeID, node := range nodes {
+		var totalCapacity btcutil.Amount
+		for _, ch := range node.Channels {
+			var policy *lndclient.RoutingPolicy
+			if ch.Node1 == nodeID && ch.Node2 == target {
+				policy = ch.Node1Policy
+			} else if ch.Node1 == target && ch.Node2 == nodeID {
+				policy = ch.Node2Policy
+			}
+
+			if policy == nil {
+				continue
+			}
+
+			totalCapacity += ch.Capacity
+
+			log.Debugf("'%v', policy=%v",
+				node.Alias, spew.Sdump(policy))
+			fee := policy.FeeBaseMsat +
+				policy.FeeRateMilliMsat*amtMsat
+
+			// For all peers we'll save the "maximum" routing fee
+			// for that peer (given the 1mm sats dummy amount).
+			if fee > maxFeePerNode[nodeID] {
+				maxFeePerNode[nodeID] = fee
+			}
+		}
+		totalCapacityPerNode[nodeID] = totalCapacity
+	}
+
+	r.nodesByMaxFee = make([]nodeFeeInfo, 0, len(maxFeePerNode))
+
+	// Sort peers by maximum fee, so that we can later or disable edges
+	// in a gradual way.
+	for nodeID, fee := range maxFeePerNode {
+		r.nodesByMaxFee = append(
+			r.nodesByMaxFee, nodeFeeInfo{
+				node:     nodeID,
+				capacity: totalCapacityPerNode[nodeID],
+				fee:      fee,
+			},
+		)
+	}
+
+	sort.Slice(r.nodesByMaxFee, func(i, j int) bool {
+		return r.nodesByMaxFee[i].fee < r.nodesByMaxFee[j].fee
+	})
+
+	for i, nodeFee := range r.nodesByMaxFee {
+		log.Tracef("nodesByMaxFee[%v] = %v (%v)", i,
+			nodeFee.node.String(), nodeFee.fee)
+	}
+
+	r.target = target
+	r.amount = amt
+
+	return nil
+}
+
+// BeforePayment will reconfigure the mission control on each payment attempt.
+func (r *lowToHighRoutingPlugin) BeforePayment(ctx context.Context,
+	currAttempt int, maxAttempts int) error {
+
+	queryRoutesReq := lndclient.QueryRoutesRequest{
+		Source:            &r.lnd.NodePubkey,
+		PubKey:            r.target,
+		AmtMsat:           lnwire.MilliSatoshi(r.amount * 1000),
+		FeeLimitMsat:      lnwire.MilliSatoshi(r.amount * 1000),
+		UseMissionControl: true,
+	}
+
+	if log.Level() == btclog.LevelTrace {
+		res, err := r.lnd.Client.QueryRoutes(ctx, queryRoutesReq)
+		log.Tracef("BeforePayment() QueryRoutes(1)=%v, err=%v",
+			spew.Sdump(res), err)
+	}
+
+	// Do not do anything unless we tried to route the payment at least
+	// once.
+	if currAttempt < 2 {
+		return nil
+	}
+
+	// Calculate the limit until we'll disable edges. Anything above this
+	// limit should be encouraged.
```

Comment:
Expand comment to explain how we're scaling this a bit? I had to manually calculate a few values here to understand how we're stepping the limit up per attempt.